### PR TITLE
Porting lts uaa version v74.5.32 that upgrades flyway to version 7 and adding mysql 8 support

### LINKED ISSUE #1799: Flyway core library requires version bump
What version of UAA are you running?
75.15.0

How are you deploying the UAA?
I am deploying the UAA

using a bosh release I downloaded from bosh.io
What did you expect to see? What goal are you trying to achieve with the UAA?
Flyway > 8.4.2

What did you see instead?
Flyway = 5.2.4

Notes:
Flyway ≤ 8.4.2 is prone to H2 Console allows remote attackers to execute arbitrary code via a jdbc:h2:mem JDBC URL. [CVE-2022-23221]
https://flywaydb.org/documentation/learnmore/releaseNotes.html#8.4.3

### LINKED ISSUE #1832: UAA does not work with MySQL 8.0.2 or higher
Hi,

I tried to deploy the UAA v75.17.0 
using cf-release 
with MySQL version 8.0.2 (AWS managed Aurora) or higher.

I got the following Exception:
`Caused by: java.sql.SQLException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups g, group_membership gm where gm.group_id = g.id and gm.identity_zone_id =' at line 1
Query is: select g.id,g.displayName,g.description,g.created,g.lastModified,g.version,g.identity_zone_id from groups g, group_membership gm where gm.group_id = g.id and gm.identity_zone_id = g.identity_zone_id and gm.identity_zone_id = ? and gm.member_id in (?);, parameters ['uaa','1b41b788-2825-4eaa-ade5-cde7caffcf75']
java thread: main
        at org.mariadb.jdbc.internal.util.LogQueryTool.exceptionWithQuery(LogQueryTool.java:163) ~[mariadb-java-client-2.3.0.jar:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:280) ~[mariadb-java-client-2.3.0.jar:?]
        at jdk.internal.reflect.GeneratedMethodAccessor71.invoke(Unknown Source) ~[?:?]`

In 2017 MySQL reserved the `groups` keyword since version 8.0.2 [(MySQL Ref, CTRL-f for groups)](https://dev.mysql.com/doc/refman/8.0/en/keywords.html).

BTW: #957 Didnt solve the problem entirely.

The table and column names should be quoted depending on the specific dbms (mysql or postgres).

### PR DESCRIPTION
**Pull request by ram-pivot**

### CODE REVIEW

**Comment by strehle** [comment]
sonar 
https://sonarcloud.io/summary/new_code?id=cloudfoundry-identity-parent&pullRequest=1950

File: server/src/main/java/org/cloudfoundry/identity/uaa/util/beans/DbUtils.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package org.cloudfoundry.identity.uaa.util.beans;
+
```
Comment (strehle) [inline server/src/main/java/org/cloudfoundry/identity/uaa/util/beans/DbUtils.java:2]: can you please check for sonar smells
https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=1950&id=cloudfoundry-identity-parent&open=AYIdJm8AY6EzI4nYYplF

Comment (ram-pivot) [reply server/src/main/java/org/cloudfoundry/identity/uaa/util/beans/DbUtils.java:2]: Sure we will check this thanks.

**Review by strehle — APPROVED** [review]
I assume you will deactivate old DBs later on ? 
e.g. https://hush-house.pivotal.io/teams/cf-uaa/pipelines/uaa-acceptance-gcp/jobs/unit-tests-postgresql-9.5

The newer already active, thanks for that.

For our side we do not rely on postgresql 9.x anymore so you can go ahead

**Comment by pabloarodas** [comment]
> I assume you will deactivate old DBs later on ? e.g. https://hush-house.pivotal.io/teams/cf-uaa/pipelines/uaa-acceptance-gcp/jobs/unit-tests-postgresql-9.5
> 
> The newer already active, thanks for that.
> 
> For our side we do not rely on postgresql 9.x anymore so you can go ahead

We still have some usages of PostgreSQL 9.x on our side, so for now we will keep it active, but you are right, we will deactivate old DBs later on.

**Review by hsinn0 — APPROVED** [review]
Approving pending the seemingly flaky unit tests to pass (We probably should look into those unit test failures.)

File: server/src/main/java/org/cloudfoundry/identity/uaa/util/beans/DbUtils.java (hunk 1 above)
Comment (pabloarodas) [reply server/src/main/java/org/cloudfoundry/identity/uaa/util/beans/DbUtils.java:2]: We fixed the security warnings and critical code smells. For DbUtils class we were able to fix 5 major code smells.

**Comment by ram-pivot** [comment]
Sonar checks passed and PR checks also passed. We also had 2 approvers. So going ahead and merging this PR.

**Pull request merged**